tests/unit/lib/SGSparseMatrix_unittest.cc

```diff
@@ -157,4 +159,110 @@ TEST(SGSparseMatrix, io_libsvm)
 			EXPECT_NEAR(m[i].features[j].entry, m_from_file[i].features[j].entry, 1E-14);
 		}
 	}
+    //remove file generated by test
+	unlink("SGSparseMatrix_io_libsvm_output.txt");
+}
+
+TEST(SGSparseMatrix, access_by_index_non_square)
+{
+    CRandom randGenerator(0);
+    float64_t sparseLevel=0.1;
+    const index_t numberOfFeatures=50;
+    const index_t numberOfVectors=100;
+
+    SGSparseMatrix<float64_t> matrix(numberOfFeatures, numberOfVectors);
+
+    for (index_t featIndex=0; featIndex<numberOfFeatures; ++featIndex)
+        for (index_t vecIndex=0; vecIndex<numberOfFeatures; ++vecIndex){
+            float64_t randomNumber=randGenerator.random(0.0,1.0);
+            if (randomNumber<=sparseLevel)
+                matrix(vecIndex,featIndex)=randomNumber;
+
+            if (randomNumber<=sparseLevel)
+                EXPECT_EQ(matrix(vecIndex,featIndex), randomNumber);
+            else
+                EXPECT_EQ(matrix(vecIndex,featIndex), 0.0);
+        }
+}
+
+TEST(SGSparseMatrix, get_transposed_more_features)
+{
+    CRandom randGenerator(0);//generate random number to control the level of sparsity
+    const float64_t sparseLevel=0.1;
+ 
+    const index_t numberOfFeatures=100;
+    const index_t numberOfVectors=50;
+    SGSparseMatrix<int32_t> sparseMatrix(numberOfFeatures, numberOfVectors);
+
+    for (index_t featIndex=0; featIndex<numberOfFeatures; ++featIndex)
+        for (index_t vecIndex=0; vecIndex<numberOfVectors; ++vecIndex){
+            float64_t randomNumber=randGenerator.random(0.0,1.0);
+            if (randomNumber<=sparseLevel)
+                sparseMatrix(vecIndex,featIndex)=randomNumber;
+        }
```

Comment:
Since you are using the same code to create the sparse matrix in several unit tests, why don't you use pull out that functionality into one function that then your tests call?